docs/codeql/codeql-overview/codeql-changelog/codeql-cli-2.20.0.rst

```diff
@@ -0,0 +1,196 @@
+
+Major Analysis Improvements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+JavaScript/TypeScript
+"""""""""""""""""""""
+
+*   The :code:`js/incomplete-sanitization` query now also checks regular expressions constructed using :code:`new RegExp(..)`. Previously it only checked regular expression literals.
+*   Regular expression-based sanitisers implemented with :code:`new RegExp(..)` are now detected in more cases.
+*   Regular expression related queries now account for unknown flags.
+
+Minor Analysis Improvements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+C#
+""
+
+*   Added support for data-flow through member accesses of objects with :code:`dynamic` types.
+*   Only extract *public* and *protected* members from reference assemblies. This yields an approximate average speed-up of around 10% for extraction and query execution. Custom MaD rows using :code:`Field`\ -based summaries may need to be changed to :code:`SyntheticField`\ -based flows if they reference private fields.
+*   Added :code:`Microsoft.AspNetCore.Components.NagivationManager::Uri` as a remote flow source, since this value may contain user-specified values.
+*   Added the following URI-parsing methods as summaries, as they may be tainted with user-specified values:
+
+    *   :code:`System.Web.HttpUtility::ParseQueryString`
+    *   :code:`Microsoft.AspNetCore.WebUtilities.QueryHelpers::ParseQuery`
+    *   :code:`Microsoft.AspNetCore.WebUtilities.QueryHelpers::ParseNullableQuery`
+    
+*   Added :code:`js-interop` sinks for the :code:`InvokeAsync` and :code:`InvokeVoidAsync` methods of :code:`Microsoft.JSInterop.IJSRuntime`, which can run arbitrary JavaScript.
+
+Golang
+""""""
+
+*   A call to a method whose name starts with "Debug", "Error", "Fatal", "Info", "Log", "Output", "Panic", "Print", "Trace", "Warn" or "With" defined on an interface whose name ends in "logger" or "Logger" is now considered a LoggerCall. In particular, it is a sink for :code:`go/clear-text-logging` and :code:`go/log-injection`. This may lead to some more alerts in those queries.
+
+Java/Kotlin
+"""""""""""
+
+*   Calling :code:`coll.contains(x)` is now a taint sanitizer (for any query) for the value :code:`x`, where :code:`coll` is a collection of constants.
+
+JavaScript/TypeScript
+"""""""""""""""""""""
+
+*   Added taint-steps for :code:`String.prototype.toWellFormed`.
+*   Added taint-steps for :code:`Map.groupBy` and :code:`Object.groupBy`.
+*   Added taint-steps for :code:`Array.prototype.findLast`.
+*   Added taint-steps for :code:`Array.prototype.findLastIndex`.
+
+Deprecated APIs
+~~~~~~~~~~~~~~~
+
+C/C++
+"""""
+
+*   The :code:`NonThrowing` class (:code:`semmle.code.cpp.models.interfaces.NonThrowing`) has been deprecated. Please use the :code:`NonCppThrowingFunction` class instead.
```

Comment:
This is not correct, we only fixed this on main. I don't think it's worth propagating the updated text everywhere, but since this hasn't been published yet we might as well:
```suggestion
*   The :code:`NonThrowingFunction` class (:code:`semmle.code.cpp.models.interfaces.NonThrowing.NonThrowingFunction`) has been deprecated. Please use the :code:`NonCppThrowingFunction` class instead.
```

Reply:
PR for reference: https://github.com/github/codeql/pull/18281